### PR TITLE
Complete node-home-work-3 as per the evaluation criteria

### PR DESCRIPTION
**Pull request by rachapathin**
1. Created different actions in stream.js **file.**
2. Reading and displaying data from text file
3. Reading data from CSVfile and converting CSVdata to JSON.
4. Reading data from CSVfile and converting CSVdata to new JSON file.
5. Bundling Multiple CSS files in ton single file(bundle.css) and appending remote CSS file content to bundle.css file

### CODE REVIEW

File: utils/streams.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    program.help();
+} else if (process.argv[2] == '-h' || process.argv[2] == '--help') {
+    program.help();       
+} else{
+    switch(program.action) {
+        case 'reverse': reverse();
+            break;
+        case 'transform': transform();
+            break;
+        case 'outputFile': outputFile();
```
Comment (AliakseiBychyk) [inline utils/streams.js]: Don't you need to pass a `program.file` or `program.path`  as an argument to the `outputFile()` and other following functions?

File: utils/streams.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        const cssFileData = fs.readFileSync(`${process.cwd()}/${program.path}/${fileName}`, "utf8");      
+        writeStream.write(cssFileData);        
+    });
+    writeStream.end();
+
+    https.get('https://epa.ms/nodejs18-hw3-css', function(res){
+        res.setEncoding('utf8');
+        res.on('data', function(cssData) {
+            fs.appendFileSync(`${program.path}/bundle.css`, cssData);
+            console.log("CSS conntent added successfully!");
```
Comment (AliakseiBychyk) [inline utils/streams.js]: Please use single quotes or configure linter appropriately.

File: utils/streams.js (hunk 1 above)
Comment (AliakseiBychyk) [reply utils/streams.js]: Also to use switch statement is not the best practice. The better approach is just define an object with properties that are defined functions and access these functions via bracket notation:

`const functions = { reverse, transform, outputFile, convertFromFile };`

`functions[program.actions]()`

File: utils/streams.js (hunk 2 above)
Comment (rachapathin) [reply utils/streams.js]: Resolved

File: utils/streams.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        case 'cssBundler': cssBundler();
+            break;
+        default: process.stdout.write('Please pass correct arguments');
+    }
+}
+
+/* Reverse string Functionality */
+function reverse () {
+    process.stdin.on('data', function (data) {
+        process.stdout.write(`Reversed String: ${data.toString().split("").reverse().join("")}`);
```
[the remark this answers is not in the document]
Comment (rachapathin) [reply utils/streams.js]: Resolved

File: utils/streams.js (hunk 1 above)
Comment (rachapathin) [reply utils/streams.js]: Resolved

**Pull request merged**